wordpress-mobile/WordPress-iOS
# let it = be(.swift4)  /// WordPress iOS is now Swift 4!

### Details:
In this PR we're bringing the main target over to the Swift 4 side. TL;DR:

- Swift entities must be explicitly exposed over to ObjectiveC by means of the **@objc** operator.
- New **NSAttributedStringKey** 'extensible enum', along with a few UIKit new native types.
- Since i was there... removed a few Implicitly Unwrapped Optionals, here and there.
- Fixed several warnings (RE: **Int** type initializer, and few others).

Closes #7343

Needs Review: @frosty @aerych @diegoreymendez @elibud @nheagy @koke @SergioEstevao @ScoutHarris 

All hands on deck, ladies and gentlemen. We'll merge after the PR gets 3 squirrels. Or more!
Thanks in advance!!

### To test:
- Review the code (as close as possible). Kaleidoscope is the perfect tool for the job!
- Verify the app builds
- Run a smoke test and verify nothing huge breaks
- Verify that the unit tests pass
